peakina/readers/csv.py

```diff
@@ -63,24 +63,31 @@ def read_csv(
     )
 
 
-def _line_count(filepath_or_buffer: "FilePathOrBuffer") -> int:
-    with open(filepath_or_buffer) as f:
+def _line_count(filepath_or_buffer: "FilePathOrBuffer", encoding: Optional[str]) -> int:
+    with open(filepath_or_buffer, encoding=encoding) as f:
         lines = 0
         buf_size = 1024 * 1024
         read_f = f.read  # loop optimization
 
         buf = read_f(buf_size)
         while buf:
+            finish_by_line_break = buf.endswith("\n")
             lines += buf.count("\n")
             buf = read_f(buf_size)
 
+        if not finish_by_line_break:
```

Comment:
I believe this might break on an empty file

Reply:
[31cc5ce](https://github.com/ToucanToco/peakina/pull/129/commits/31cc5ce6c3a66bd6bfa59a7b3798d2a7c734ee5b)